### DISCUSSION TITLE
Tuning LightGBM model needs too much memory...

### DISCUSSION THREAD

**Discussion by Goosang-Yu** [pycaret/pycaret#2592] [Q&A]
Hi guys, I have some trouble when I try to tune my LightGBM
CatBoost or XGBoost tuning works well,
But only LGBM tuning shows error like this:

![image (6)](https://user-images.githubusercontent.com/67215983/170612081-681aa616-8d1b-41eb-abbf-0834cec8f11a.png)

```
Fitting 5 folds for each of 10 candidates, totalling 50 fits
[Parallel(n_jobs=-1)]: Using backend LokyBackend with 32 concurrent workers.
Canceled future for execute_request message before replies were done
The Kernel crashed while executing code in the the current cell or a previous cell. Please review the code in the cell(s) to identify a possible cause of the failure. Click here for more info. View Jupyter log for further details.
```

![image (7)](https://user-images.githubusercontent.com/67215983/170612196-fccc69c8-ee83-44ea-96d8-7ca62ece24b8.png)

I think LGBM tuning needs too much memory...
Is there any way to ues tune_model() without additional RAM?

Thanks!

**Comment by celestinoxp** [comment]
version of pycaret and lightgbm?

**Comment by Goosang-Yu** [comment]
Hi Celestino,

Version:
pycaret=2.3.10
lgbm=3.3.2

And I attach the list of all packages.

Goosang


2022년 5월 27일 (금) 오후 5:27, CelestinoXP ***@***.***>님이 작성:

> version of pycaret and lightgbm?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pycaret/pycaret/discussions/2592#discussioncomment-2832685>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQA2E32JNRPK5N2VEUFOMYLVMCBOLANCNFSM5XC275FQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

# packages in environment at /home/hkim/anaconda2/envs/pycaret:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
adagio                    0.2.3                    pypi_0    pypi
aiohttp                   3.8.1                    pypi_0    pypi
aiosignal                 1.2.0                    pypi_0    pypi
alembic                   1.7.7                    pypi_0    pypi
analytics-python          1.4.0                    pypi_0    pypi
ansi2html                 1.7.0                    pypi_0    pypi
antlr4-python3-runtime    4.9.3                    pypi_0    pypi
anyio                     3.5.0                    pypi_0    pypi
appdirs                   1.4.4                    pypi_0    pypi
argon2-cffi               21.3.0                   pypi_0    pypi
argon2-cffi-bindings      21.2.0                   pypi_0    pypi
asgiref                   3.5.0                    pypi_0    pypi
asttokens                 2.0.5                    pypi_0    pypi
async-timeout             4.0.2                    pypi_0    pypi
attrs                     21.4.0                   pypi_0    pypi
autopage                  0.5.0                    pypi_0    pypi
autoviz                   0.1.38                   pypi_0    pypi
azure-core                1.23.1                   pypi_0    pypi
azure-storage-blob        12.11.0                  pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
backoff                   1.10.0                   pypi_0    pypi
bcrypt                    3.2.0                    pypi_0    pypi
beautifulsoup4            4.11.1                   pypi_0    pypi
biopython                 1.79             py38h497a2fe_1    conda-forge
bleach                    5.0.0                    pypi_0    pypi
blis                      0.7.7                    pypi_0    pypi
bokeh                     2.4.2                    pypi_0    pypi
boruta                    0.3                      pypi_0    pypi
boto3                     1.21.42                  pypi_0    pypi
botocore                  1.24.42                  pypi_0    pypi
brotli                    1.0.9                    pypi_0    pypi
ca-certificates           2020.10.14                    0    anaconda
cachetools                5.0.0                    pypi_0    pypi
catalogue                 1.0.0                    pypi_0    pypi
catboost                  1.0.5                    pypi_0    pypi
certifi                   2021.10.8                pypi_0    pypi
cffi                      1.15.0                   pypi_0    pypi
charset-normalizer        2.0.12                   pypi_0    pypi
chart-studio              1.1.0                    pypi_0    pypi
ciso8601                  2.2.0                    pypi_0    pypi
click                     8.1.2                    pypi_0    pypi
cliff                     3.10.1                   pypi_0    pypi
cloudpickle               2.0.0                    pypi_0    pypi
cmaes                     0.8.2                    pypi_0    pypi
cmd2                      2.4.1                    pypi_0    pypi
colorcet                  3.0.0                    pypi_0    pypi
colorlog                  6.6.0                    pypi_0    pypi
colorlover                0.3.0                    pypi_0    pypi
colour                    0.1.5                    pypi_0    pypi
cryptography              36.0.2                   pypi_0    pypi
cudatoolkit               10.2.89              hfd86e86_1    anaconda
cudnn                     7.6.5                cuda10.2_0    anaconda
cufflinks                 0.17.3                   pypi_0    pypi
cycler                    0.11.0                   pypi_0    pypi
cymem                     2.0.6                    pypi_0    pypi
dash                      2.3.1                    pypi_0    pypi
dash-auth                 1.4.1                    pypi_0    pypi
dash-bootstrap-components 0.13.1                   pypi_0    pypi
dash-core-components      2.0.0                    pypi_0    pypi
dash-cytoscape            0.3.0                    pypi_0    pypi
dash-html-components      2.0.0                    pypi_0    pypi
dash-table                5.0.0                    pypi_0    pypi
databricks-cli            0.16.6                   pypi_0    pypi
dataclasses               0.6                      pypi_0    pypi
debugpy                   1.6.0                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
dill                      0.3.4                    pypi_0    pypi
distlib                   0.3.4                    pypi_0    pypi
docker                    5.0.3                    pypi_0    pypi
docker-pycreds            0.4.0                    pypi_0    pypi
dtreeviz                  1.3.5                    pypi_0    pypi
emoji                     1.7.0                    pypi_0    pypi
entrypoints               0.4                      pypi_0    pypi
evidently                 0.1.48.dev0              pypi_0    pypi
executing                 0.8.3                    pypi_0    pypi
explainerdashboard        0.3.8.2                  pypi_0    pypi
fairlearn                 0.7.0                    pypi_0    pypi
fastapi                   0.75.2                   pypi_0    pypi
fastjsonschema            2.15.3                   pypi_0    pypi
fasttreeshap              0.1.1                    pypi_0    pypi
ffmpy                     0.3.0                    pypi_0    pypi
filelock                  3.6.0                    pypi_0    pypi
flask                     2.1.1                    pypi_0    pypi
flask-compress            1.11                     pypi_0    pypi
flask-seasurf             1.1.1                    pypi_0    pypi
flask-simplelogin         0.1.1                    pypi_0    pypi
flask-wtf                 0.15.1                   pypi_0    pypi
fonttools                 4.32.0                   pypi_0    pypi
frozenlist                1.3.0                    pypi_0    pypi
fs                        2.4.15                   pypi_0    pypi
fsspec                    2022.3.0                 pypi_0    pypi
fugue                     0.6.6                    pypi_0    pypi
funcy                     1.17                     pypi_0    pypi
future                    0.18.2                   pypi_0    pypi
gensim                    3.8.3                    pypi_0    pypi
gevent                    21.12.0                  pypi_0    pypi
gitdb                     4.0.9                    pypi_0    pypi
gitpython                 3.1.27                   pypi_0    pypi
google-api-core           2.7.2                    pypi_0    pypi
google-auth               2.6.5                    pypi_0    pypi
google-cloud-core         2.3.0                    pypi_0    pypi
google-cloud-storage      2.3.0                    pypi_0    pypi
google-crc32c             1.3.0                    pypi_0    pypi
google-resumable-media    2.3.2                    pypi_0    pypi
googleapis-common-protos  1.56.0                   pypi_0    pypi
gradio                    2.9.4                    pypi_0    pypi
greenlet                  1.1.2                    pypi_0    pypi
grpcio                    1.43.0                   pypi_0    pypi
gunicorn                  20.1.0                   pypi_0    pypi
h11                       0.13.0                   pypi_0    pypi
holoviews                 1.14.8                   pypi_0    pypi
htmlmin                   0.1.12                   pypi_0    pypi
hvplot                    0.7.3                    pypi_0    pypi
hyperopt                  0.2.7                    pypi_0    pypi
idna                      3.3                      pypi_0    pypi
imagehash                 4.2.1                    pypi_0    pypi
imageio                   2.15.0                   pypi_0    pypi
imbalanced-learn          0.7.0                    pypi_0    pypi
importlib-metadata        4.11.3                   pypi_0    pypi
importlib-resources       5.7.1                    pypi_0    pypi
iniconfig                 1.1.1                    pypi_0    pypi
interpret                 0.2.4                    pypi_0    pypi
interpret-core            0.2.7                    pypi_0    pypi
ipykernel                 6.13.0                   pypi_0    pypi
ipython                   8.2.0                    pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
ipywidgets                7.7.0                    pypi_0    pypi
isodate                   0.6.1                    pypi_0    pypi
itsdangerous              2.1.2                    pypi_0    pypi
jedi                      0.18.1                   pypi_0    pypi
jinja2                    3.1.1                    pypi_0    pypi
jmespath                  1.0.0                    pypi_0    pypi
joblib                    1.0.1                    pypi_0    pypi
jsonschema                4.4.0                    pypi_0    pypi
jupyter                   1.0.0                    pypi_0    pypi
jupyter-client            7.2.2                    pypi_0    pypi
jupyter-console           6.4.3                    pypi_0    pypi
jupyter-core              4.9.2                    pypi_0    pypi
jupyter-dash              0.4.2                    pypi_0    pypi
jupyterlab-pygments       0.2.2                    pypi_0    pypi
jupyterlab-widgets        1.1.0                    pypi_0    pypi
kiwisolver                1.4.2                    pypi_0    pypi
kmodes                    0.12.1                   pypi_0    pypi
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
libblas                   3.9.0           14_linux64_openblas    conda-forge
libcblas                  3.9.0           14_linux64_openblas    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 11.2.0              h1d223b6_15    conda-forge
libgfortran-ng            11.2.0              h69a702a_15    conda-forge
libgfortran5              11.2.0              h5c6108e_15    conda-forge
libgomp                   11.2.0              h1d223b6_15    conda-forge
liblapack                 3.9.0           14_linux64_openblas    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libopenblas               0.3.20          pthreads_h78a6416_0    conda-forge
libstdcxx-ng              11.2.0              he4da1e4_15    conda-forge
libzlib                   1.2.11            h166bdaf_1014    conda-forge
lightgbm                  3.3.2                    pypi_0    pypi
lime                      0.2.0.1                  pypi_0    pypi
linkify-it-py             1.0.3                    pypi_0    pypi
llvmlite                  0.37.0                   pypi_0    pypi
m2cgen                    0.9.0                    pypi_0    pypi
mako                      1.2.0                    pypi_0    pypi
markdown                  3.3.6                    pypi_0    pypi
markdown-it-py            2.1.0                    pypi_0    pypi
markupsafe                2.0.1                    pypi_0    pypi
matplotlib                3.5.1                    pypi_0    pypi
matplotlib-inline         0.1.3                    pypi_0    pypi
mdit-py-plugins           0.3.0                    pypi_0    pypi
mdurl                     0.1.1                    pypi_0    pypi
missingno                 0.5.1                    pypi_0    pypi
mistune                   0.8.4                    pypi_0    pypi
mlflow                    1.25.1                   pypi_0    pypi
mlxtend                   0.19.0                   pypi_0    pypi
monotonic                 1.6                      pypi_0    pypi
msgpack                   1.0.3                    pypi_0    pypi
msrest                    0.6.21                   pypi_0    pypi
multidict                 6.0.2                    pypi_0    pypi
multimethod               1.8                      pypi_0    pypi
multiprocess              0.70.12.2                pypi_0    pypi
murmurhash                1.0.6                    pypi_0    pypi
nbclient                  0.6.0                    pypi_0    pypi
nbconvert                 6.5.0                    pypi_0    pypi
nbformat                  5.3.0                    pypi_0    pypi
ncurses                   6.3                  h27087fc_1    conda-forge
nest-asyncio              1.5.5                    pypi_0    pypi
networkx                  2.8                      pypi_0    pypi
nltk                      3.7                      pypi_0    pypi
notebook                  6.4.10                   pypi_0    pypi
numba                     0.54.1                   pypi_0    pypi
numexpr                   2.8.1                    pypi_0    pypi
numpy                     1.19.5                   pypi_0    pypi
oauthlib                  3.2.0                    pypi_0    pypi
openssl                   3.0.2                h166bdaf_1    conda-forge
optuna                    2.10.0                   pypi_0    pypi
orjson                    3.6.8                    pypi_0    pypi
oyaml                     1.0                      pypi_0    pypi
packaging                 21.3                     pypi_0    pypi
pandas                    1.4.2                    pypi_0    pypi
pandas-profiling          3.1.0                    pypi_0    pypi
pandocfilters             1.5.0                    pypi_0    pypi
panel                     0.12.7                   pypi_0    pypi
param                     1.12.1                   pypi_0    pypi
paramiko                  2.10.3                   pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
pathos                    0.2.8                    pypi_0    pypi
pathtools                 0.1.2                    pypi_0    pypi
patsy                     0.5.2                    pypi_0    pypi
pbr                       5.8.1                    pypi_0    pypi
pexpect                   4.8.0                    pypi_0    pypi
phik                      0.12.2                   pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    9.1.0                    pypi_0    pypi
pip                       22.0.4             pyhd8ed1ab_0    conda-forge
plac                      1.1.3                    pypi_0    pypi
platformdirs              2.5.1                    pypi_0    pypi
plotly                    5.7.0                    pypi_0    pypi
pluggy                    1.0.0                    pypi_0    pypi
pox                       0.3.0                    pypi_0    pypi
ppft                      1.6.6.4                  pypi_0    pypi
preshed                   3.0.6                    pypi_0    pypi
prettytable               3.2.0                    pypi_0    pypi
prometheus-client         0.14.1                   pypi_0    pypi
prometheus-flask-exporter 0.20.1                   pypi_0    pypi
promise                   2.3                      pypi_0    pypi
prompt-toolkit            3.0.29                   pypi_0    pypi
protobuf                  3.20.0                   pypi_0    pypi
psutil                    5.9.0                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
py                        1.11.0                   pypi_0    pypi
py4j                      0.10.9.5                 pypi_0    pypi
pyaml                     21.10.1                  pypi_0    pypi
pyarrow                   7.0.0                    pypi_0    pypi
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pycaret                   2.3.10                   pypi_0    pypi
pycparser                 2.21                     pypi_0    pypi
pycryptodome              3.14.1                   pypi_0    pypi
pyct                      0.4.8                    pypi_0    pypi
pydantic                  1.9.0                    pypi_0    pypi
pydub                     0.25.1                   pypi_0    pypi
pygments                  2.11.2                   pypi_0    pypi
pyjwt                     2.3.0                    pypi_0    pypi
pyldavis                  3.2.2                    pypi_0    pypi
pynacl                    1.5.0                    pypi_0    pypi
pynndescent               0.5.6                    pypi_0    pypi
pyod                      0.9.9                    pypi_0    pypi
pyparsing                 3.0.8                    pypi_0    pypi
pyperclip                 1.8.2                    pypi_0    pypi
pyrsistent                0.18.1                   pypi_0    pypi
pytest                    7.1.1                    pypi_0    pypi
python                    3.8.12          h0744224_3_cpython    conda-forge
python-dateutil           2.8.2                    pypi_0    pypi
python-graphviz           0.20                     pypi_0    pypi
python-multipart          0.0.5                    pypi_0    pypi
python_abi                3.8                      2_cp38    conda-forge
pytz                      2022.1                   pypi_0    pypi
pyviz-comms               2.2.0                    pypi_0    pypi
pywavelets                1.3.0                    pypi_0    pypi
pyyaml                    5.4.1                    pypi_0    pypi
pyzmq                     22.3.0                   pypi_0    pypi
qpd                       0.2.6                    pypi_0    pypi
qtconsole                 5.3.0                    pypi_0    pypi
qtpy                      2.0.1                    pypi_0    pypi
querystring-parser        1.2.4                    pypi_0    pypi
ray                       1.12.0                   pypi_0    pypi
readline                  8.1                  h46c0cb4_0    conda-forge
regex                     2022.3.15                pypi_0    pypi
requests                  2.27.1                   pypi_0    pypi
requests-oauthlib         1.3.1                    pypi_0    pypi
retrying                  1.3.3                    pypi_0    pypi
rsa                       4.8                      pypi_0    pypi
s3transfer                0.5.2                    pypi_0    pypi
salib                     1.4.5                    pypi_0    pypi
scikit-image              0.19.2                   pypi_0    pypi
scikit-learn              0.23.2                   pypi_0    pypi
scikit-optimize           0.9.0                    pypi_0    pypi
scikit-plot               0.3.7                    pypi_0    pypi
scipy                     1.5.4                    pypi_0    pypi
seaborn                   0.11.2                   pypi_0    pypi
send2trash                1.8.0                    pypi_0    pypi
sentry-sdk                1.5.10                   pypi_0    pypi
setproctitle              1.2.3                    pypi_0    pypi
setuptools                60.10.0                  pypi_0    pypi
shap                      0.40.0                   pypi_0    pypi
shortuuid                 1.0.8                    pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
skope-rules               1.0.1                    pypi_0    pypi
slicer                    0.0.7                    pypi_0    pypi
smart-open                5.2.1                    pypi_0    pypi
smmap                     5.0.0                    pypi_0    pypi
sniffio                   1.2.0                    pypi_0    pypi
soupsieve                 2.3.2.post1              pypi_0    pypi
spacy                     2.3.7                    pypi_0    pypi
sqlalchemy                1.4.35                   pypi_0    pypi
sqlite                    3.38.2               h4ff8645_0    conda-forge
sqlparse                  0.4.2                    pypi_0    pypi
srsly                     1.0.5                    pypi_0    pypi
stack-data                0.2.0                    pypi_0    pypi
starlette                 0.17.1                   pypi_0    pypi
statsmodels               0.13.2                   pypi_0    pypi
stevedore                 3.5.0                    pypi_0    pypi
tabulate                  0.8.9                    pypi_0    pypi
tangled-up-in-unicode     0.1.0                    pypi_0    pypi
tenacity                  8.0.1                    pypi_0    pypi
tensorboardx              2.5                      pypi_0    pypi
terminado                 0.13.3                   pypi_0    pypi
textblob                  0.17.1                   pypi_0    pypi
thinc                     7.4.5                    pypi_0    pypi
threadpoolctl             3.1.0                    pypi_0    pypi
tifffile                  2022.4.8                 pypi_0    pypi
tinycss2                  1.1.1                    pypi_0    pypi
tk                        8.6.12               h27826a3_0    conda-forge
tomli                     2.0.1                    pypi_0    pypi
tornado                   6.1                      pypi_0    pypi
tqdm                      4.64.0                   pypi_0    pypi
traitlets                 5.1.1                    pypi_0    pypi
treeinterpreter           0.2.3                    pypi_0    pypi
triad                     0.6.0                    pypi_0    pypi
tune-sklearn              0.4.2                    pypi_0    pypi
typing-extensions         4.2.0                    pypi_0    pypi
ua-parser                 0.10.0                   pypi_0    pypi
uc-micro-py               1.0.1                    pypi_0    pypi
umap-learn                0.5.3                    pypi_0    pypi
urllib3                   1.26.9                   pypi_0    pypi
uvicorn                   0.17.6                   pypi_0    pypi
virtualenv                20.14.1                  pypi_0    pypi
visions                   0.7.4                    pypi_0    pypi
waitress                  2.1.1                    pypi_0    pypi
wandb                     0.12.14                  pypi_0    pypi
wasabi                    0.9.1                    pypi_0    pypi
wcwidth                   0.2.5                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
websocket-client          1.3.2                    pypi_0    pypi
werkzeug                  2.0.3                    pypi_0    pypi
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
widgetsnbextension        3.6.0                    pypi_0    pypi
wordcloud                 1.8.1                    pypi_0    pypi
wtforms                   3.0.1                    pypi_0    pypi
xgboost                   1.6.0                    pypi_0    pypi
xlrd                      2.0.1                    pypi_0    pypi
xz                        5.2.5                h516909a_1    conda-forge
yarl                      1.7.2                    pypi_0    pypi
yellowbrick               1.3.post1                pypi_0    pypi
zipp                      3.8.0                    pypi_0    pypi
zlib                      1.2.11            h166bdaf_1014    conda-forge
zope-event                4.5.0                    pypi_0    pypi
zope-interface            5.4.0                    pypi_0    pypi

**Comment by celestinoxp** [comment]
Can you send the size of dataset and config in setup?

**Comment by Goosang-Yu** [comment]
Transformed Train Set = (259910, 2956)
Transformed Test Set  = (28883, 2956)

Because the file size of config by save_config is about 6Gb, I can't send
it to you, sorry.

My set up is:

s = setup(data=df_train, test_data=df_test, ignore_features=list_ignore,
          target = 'Measured_PE_efficiency',
          normalize=True,
          fold=5, session_id=220413, fold_groups=df_fold_group,
          categorical_features=list_cat_feat,
numeric_features=list_num_feat,
          log_experiment='wandb', experiment_name=exp_name,
          use_gpu=True, preprocess=True, silent=True)



2022년 5월 27일 (금) 오후 8:27, CelestinoXP ***@***.***>님이 작성:

**Reply by celestinoxp** [reply]
In my opinion, pycaret or lightgbm are working correctly. I believe the problem is in the dataset.
The dataset is too big, it has too many columns (attributes)... try reducing some columns...

**Reply by BradKML** [reply]
The central issue is that LightGBM can be used for feature selection (eliminating columns before use in other models) so this have to work somehow... such that people can further refine their data. Chicken and the egg type problem

**Comment by Goosang-Yu** [comment]
Thank you for your opinion!

Does lgbm basically consume lots of memory?
XGboost, CatBoost, and other regression models trained by the same dataset
didn't show any problems..

Anyway... thanks a lot!!


2022년 5월 27일 (금) 오후 10:28, CelestinoXP ***@***.***>님이 작성:

**Reply by celestinoxp** [reply]
@Goosang-Yu The problem may be related to the depth of the trees. I saw this somewhere that I can't find now... try controlling the depth variable manually and see if the problem continues or not...

**Reply by celestinoxp** [reply]
I checked your post again and your package versions, including pycaret, are out of date!
Update to the latest version of pycaret!

**Reply by BradKML** [reply]
@celestinoxp seconding this for @Goosang-Yu and I think this happens to feature selection before using any model as well (notes from elsewhere)
* The big three are likely `num_leaves` (defaults to 31 but should go lower), `max_bin`  (start with 63 or 127 especially for wide datasets), `min_data_in_leaf` (>1000 for larger dataset)
* Other params like `feature_fraction` (between 0.3 and 0.7), `bagging_fraction` (around 0.7), `bagging_freq` (keep at 1), and `max_depth` (between 5 and 10) being also somewhat important
* For accelerating convergence, set `n_estimators` to 10000 (larger than default), `early_stopping_rounds` to 100, and `learning_rate` to 0.1 (or lower)
* Also it is recommended to set `gpu_use_dp` to false and `force_row_wise` to true
* Make sure to use `categorical_feature` baked into the library instead of OneHotEncoder
